### ISSUE TITLE
Websocket unexpected disconnect with websocket v1.11

### ISSUE THREAD

**Issue by millejoh** [millejoh/emacs-ipython-notebook#559]
**TL;DR - Don't upgrade to emacs-websocket v1.11. If you do and start seeing these behaviors the current workaround is to hand install websocket v1.9 or earlier.**

Documenting for myself and as a general query in case others are seeing the same issue.

For the past couple weeks I have been seeing issues with ein spontaneously disconnect the websocket session to a running kernel. This appears to happen randomly whenever trying to execute code, though it most reliably occurs when first opening a notebook and injecting the pytools code (i.e. the contents of `ein_remote_safe.py`).

I am seeing this both under my WSL/Ubuntu 18.04 and Ubuntu 18.10 setups. It is also affecting travis (uh oh). This does not appear to be an issue with jupyter as I have tried various version of the jupyter notebook server - 5.7.8, 5.7.7, 5.6.0 and all experience the websocket disconnects.

This is affecting both my spacemacs config and vanilla emacs as launched via: `cask emacs -Q -L ./lisp` from the directory where I have ein installed on my local machine.

~Oddly enough my Windows 10 setup is not experiencing this issue.~ Turns out I had not updated my packages in a while, as soon as I installed websocket v1.11 I started seeing the same behavior in Windows

I've tested with python and R kernels and both experience disconnects.

The logs are not very useful, other than normal communication traffic all I see in the websocket debug buffer is a `[WS] State change to connection broken by remote peer` just before the connection is lost.

Websocket was recently updated to 1.11 on June 9th along with some commits to [improve performance](https://github.com/ahyatt/emacs-websocket/commit/ff566c4ad7539e5f220772bca1977d859cee72ef). 

Next step - see if this problem persists with versions prior to websocket 1.11

**Comment by millejoh** [comment]
Okay - I tried with an earlier version of websocket (from commit ahyatt/emacs-websocket@0d96ba2ff5a25c6cd6c66f417cc9b5f38a4308ba) and the disconnects appear to have stopped (or at least are much less frequent), so this appears to be a problem with the websocket package.
